sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb

```diff
@@ -0,0 +1,227 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using Amazon Elastic Inference with Neo-compiled TensorFlow model on SageMaker\n",
+    "\n",
+    "This notebook demonstrates how to compile a pre-trained TensorFlow model using Amazon SageMaker Neo, and deploy this compiled model on SageMaker with Elastic Inference.\n",
+    "\n",
+    "Amazon Elastic Inference (EI) allows you to add inference acceleration to an Amazon SageMaker hosted endpoint for a fraction of the cost of using a full GPU instance. Running Neo-compiled models on EI provides a performance boost by optimizing the model to produce low latency inferences. This would increase inference throughput and further reduce costs. For more information please visit: https://docs.aws.amazon.com/sagemaker/latest/dg/ei.html\n",
+    "\n",
+    "This notebook is an adaption of the [Deploy pre-trained TensorFlow model to SageMaker with Elastic Inference](https://github.com/aws/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_serving_pretrained_model_elastic_inference.ipynb) notebook, with modifications showing the changes needed to deploy Neo-compiled models on SageMaker with EI.\n",
+    "\n",
+    "For this example, we will use the SageMaker Python SDK, which makes it easy to compile and deploy your model on SageMaker. \n",
+    "\n",
+    "1. [Set up the environment](#Set-up-the-environment)\n",
+    "1. [Get pre-trained model for compilation](#Get-pre-trained-model-for-compilation)\n",
+    "    1. [Import ResNet50 model from Keras](#Import-ResNet50-model-from-Keras)\n",
+    "    1. [Upload model artifact to S3 bucket](#Upload-model-artifact-to-S3-bucket)\n",
+    "1. [Compile model for EI accelerator using Neo](#Compile-model-for-EI-accelerator-using-Neo)\n",
+    "1. [Deploy compiled model to SageMaker Endpoint with EI accelerator attached](#Deploy-compiled-model-to-SageMaker-Endpoint-with-EI-accelerator-attached)\n",
+    "1. [Make an inference request to the endpoint](#Make-an-inference-request-to-the-endpoint)\n",
+    "1. [Delete the endpoint](#Delete-the-endpoint)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up the environment\n",
+    "Let's start by creating a SageMaker session and specifying:\n",
+    "\n",
+    "* The S3 bucket that you want to use for model data. This should be within the same region as the Notebook Instance, Neo compilation, and SageMaker hosting.\n",
+    "* The IAM role arn used to give compilation and hosting access to your data. See the [documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html) for how to create these. Note, if more than one role is required for notebook instances, compilation, and hosting, please replace the sagemaker.get_execution_role() with a the appropriate full IAM role arn string(s)."
```

Comment:
Bold face the note, similar to what you did when you wrote "Important: If the following..." below in this notebook.